Frontend/ui-driver/src/Screens/DriverReferralScreen/Controller.purs

```diff
@@ -0,0 +1,116 @@
+import Helpers.Utils (incrementValueOfLocalStoreKey)
+import Components.PrimaryButton as PrimaryButton
+import Common.Types.App (ShareImageConfig)
+import Engineering.Helpers.Commons (getNewIDWithTag)
+import Data.Maybe (Maybe(..), fromMaybe, isJust)
+import MerchantConfig.Utils (getMerchant, Merchant(..))
+import Common.Types.App (LazyCheck(..))
+
+instance showAction :: Show Action where
+  show _ = ""
+instance loggableAction :: Loggable Action where
+  performLog action appId = case action of
+    AfterRender -> trackAppScreenRender appId "screen" "DriverReferralScreen"
+    BackPressed -> trackAppBackPress appId (getScreen HOME_SCREEN)
+    GenericHeaderActionController act -> case act of
+      GenericHeader.PrefixImgOnClick -> do
+        trackAppActionClick appId (getScreen REFERRAL_SCREEN) "generic_header_action" "back_icon"
+        trackAppEndScreen appId (getScreen REFERRAL_SCREEN)
+      GenericHeader.SuffixImgOnClick -> trackAppActionClick appId (getScreen REFERRAL_SCREEN) "generic_header_action" "forward_icon"
+    ShowQRCode -> pure unit
+    ShareOptions -> pure unit
+    BottomNavBarAction (BottomNavBar.OnNavigate item) -> do
+      trackAppActionClick appId (getScreen REFERRAL_SCREEN) "bottom_nav_bar" "on_navigate"
+      trackAppEndScreen appId (getScreen REFERRAL_SCREEN)
+    LearnMore -> pure unit
+    PrimaryButtonActionController state act -> case act of
+      PrimaryButton.OnClick -> do
+        trackAppActionClick appId (getScreen CHOOSE_LANGUAGE_SCREEN) "primary_button_action" "next_on_click"
+        trackAppEndScreen appId (getScreen CHOOSE_LANGUAGE_SCREEN)
+      PrimaryButton.NoAction -> trackAppActionClick appId (getScreen CHOOSE_LANGUAGE_SCREEN) "primary_button_action" "no_action"
+    ReferredDriversAPIResponseAction val -> pure unit
+
+data Action = BackPressed
+            | AfterRender
+            | GenericHeaderActionController GenericHeader.Action
+            | ShowQRCode
+            | ShareOptions
+            | BottomNavBarAction BottomNavBar.Action
+            | LearnMore
+            | PrimaryButtonActionController DriverReferralScreenState PrimaryButton.Action
+            | ReferredDriversAPIResponseAction Int
+
+
+data ScreenOutput = GoToHomeScreen DriverReferralScreenState
+                  | GoToRidesScreen DriverReferralScreenState
+                  | GoToProfileScreen DriverReferralScreenState
+                  | GoToNotifications DriverReferralScreenState
+                  | SubscriptionScreen DriverReferralScreenState
+                  | GoToDriverContestScreen DriverReferralScreenState
+                  | GoBack
+
+
+eval :: Action -> DriverReferralScreenState -> Eval Action ScreenOutput DriverReferralScreenState
+
+eval BackPressed state = exit $ GoBack
+
+eval (GenericHeaderActionController (GenericHeader.PrefixImgOnClick)) state = continueWithCmd state [do pure BackPressed]
```

Comment:
Use this  `eval (GenericHeaderActionController (GenericHeader.PrefixImgOnClick)) state = exit $ GoBack`

No need of continueWithCmd